### DISCUSSION TITLE
How do I support go 1.19 updates with renovate?

### DISCUSSION THREAD

**Discussion by sheldonhull** [renovatebot/renovate#18514] [Request Help]
[Related](https://github.com/renovatebot/renovate/issues/13727)

I'm not certain if there is a work around for `gomodtidy` and other go automated artifact updates with Go 1.19 right now. 

I've tried the docker binary install option but this didn't help and honestly I'm just guessing at this point 😃

Is it possible to manage go 1.19 with renovate currently with any set of options and if not what issue should I follow for resolution?

**Comment by Gabriel-Ladzaretti** [comment]
Have you tried `binarySource=docker`[^1] instead?

> When Renovate is running using binarySource=docker (such as in the hosted Mend Renovate app) then it will pick the latest compatible version of Go to run, i.e. the latest 1.x release. Even if the go.mod has a version like go 1.14, Renovate will treat it as a ^1.14 constraint and not =1.14.

[^1]: https://docs.renovatebot.com/modules/manager/gomod/#additional-information.

**Reply by sheldonhull** [reply]
Yes. From that related thread. I set that but in believe I still get the failure to update artifact as Go mod tidy is only supported by 1.18. I don't think it recognized 1.19. Since I'm using mend and not a renovate cli into invoke I'm not certain how to get debug output to confirm.

**Reply by Gabriel-Ladzaretti** [reply]
if the PR was created by the app, the PR footer should contain a link to the log file.
this - https://app.renovatebot.com/dashboard might help as well

![image](https://user-images.githubusercontent.com/97394622/197789094-cb0b60b1-24e2-4c59-8b58-1ee9f04832f8.png)[^1]

[^1]: https://github.com/renovatebot/tutorial#part-3---getting-to-know-renovates-update-prs.

**Reply by sheldonhull** [reply]
The dashboard seems only show private repos but not any of my org repos. I'm using Mend SaaS and having it run renovate in case that matters. 

I don't see that footer at all.

**Reply by Gabriel-Ladzaretti** [reply]
Renovate under Mend SaaS should support `go 1.19.1`, therefore you should probably contact Mend support directly for further inspecting your issue.

**Reply by sheldonhull** [reply]
Running back into this.  Docker as binary source has been working great. However 1.20 is failing and after pushing 1.19 it also fails saying go mod tidy is only supported by 1.18. 

Any idea on where I should start troubleshooting?

**Reply by rarkins** [reply]
Can you copy paste the full command and error message you receive?

**Reply by sheldonhull** [reply]
I opted to try and use binarysource=global, and parsed from go.mod. This seemed to work. For now I'm good so if I get stuck again I'll follow-up and post more details. Thanks for the response!
